File: agent-client/src/types/content.ts
export interface FileInfo {
  filename: string
  size: number
  created_at: number
  modified_at: number
  path: string
}

export interface FileListResponse {
  files: FileInfo[]
}

export interface DeleteFileResponse {
  success: boolean
  message: string
}

export interface UploadRequest {
  collection_name?: string
}

export interface UploadResponse {
  task_id: string
  status: string
  filename: string
}

export interface TaskStatusResponse {
  task_id: string
  status: string
  filename?: string
  step?: string
  chunks?: number
  error?: string
  created_at?: string
}

export interface QueryRequest {
  question: string
  top_k?: number
  collection_name?: string
}

export interface SourceDocument {
  content: string
  source: string
  page?: number
  similarity?: number
}

export interface QueryResponse {
  answer: string
  sources: SourceDocument[]
  processing_time_ms: number
}

export interface ReportRequest {
  report_type: string
  parameters: Record<string, unknown>
  collection_name?: string
}

export interface ReportReference {
  standard_name: string
  section?: string
  content?: string
}

export interface ReportResponse {
  content: string
  references: ReportReference[]
  generated_at: string
}

File: agent-server/schemas/content.py
from typing import Any, Optional

from pydantic import BaseModel, Field


class UploadResponse(BaseModel):
    task_id: str
    status: str
    filename: str


class TaskStatusResponse(BaseModel):
    task_id: str
    status: str
    filename: Optional[str] = None
    step: Optional[str] = None
    chunks: Optional[int] = None
    error: Optional[str] = None
    created_at: Optional[str] = None


class FileInfo(BaseModel):
    filename: str
    size: int
    created_at: float
    modified_at: float
    path: str


class QueryRequest(BaseModel):
    question: str = Field(..., min_length=1, max_length=2000)
    top_k: int = Field(5, ge=1, le=10)
    collection_name: Optional[str] = None


class SourceDocument(BaseModel):
    content: str
    source: str
    page: Optional[int] = None
    similarity: Optional[float] = None


class QueryResponse(BaseModel):
    answer: str
    sources: list[SourceDocument]
    processing_time_ms: int


class ReportRequest(BaseModel):
    report_type: str = Field(..., min_length=1, max_length=100)
    parameters: dict[str, Any]
    collection_name: Optional[str] = None


class ReportReference(BaseModel):
    standard_name: str
    section: Optional[str] = None
    content: Optional[str] = None


class ReportResponse(BaseModel):
    content: str
    references: list[ReportReference]
    generated_at: str
